domains/client/domain-executor/src/tests.rs

```diff
@@ -1095,3 +1051,95 @@ async fn existing_bundle_can_be_resubmitted_to_new_fork() {
         Err(err) => panic!("Unexpected error: {err}"),
     }
 }
+
+#[substrate_test_utils::test(flavor = "multi_thread")]
+async fn test_cross_domains_message_should_work() {
+    let directory = TempDir::new().expect("Must be able to create temporary directory");
+
+    let mut builder = sc_cli::LoggerBuilder::new("");
+    builder.with_colors(false);
+    let _ = builder.init();
+
+    let tokio_handle = tokio::runtime::Handle::current();
+
+    // Start Ferdie
+    let mut ferdie = MockPrimaryNode::run_mock_primary_node(
+        tokio_handle.clone(),
+        Ferdie,
+        BasePath::new(directory.path().join("ferdie")),
+    );
+
+    // Run Alice (a system domain authority node)
+    let mut alice = domain_test_service::SystemDomainNodeBuilder::new(
+        tokio_handle.clone(),
+        Alice,
+        BasePath::new(directory.path().join("alice")),
+    )
+    .run_relayer()
+    .build_with_mock_primary_node(Role::Authority, &mut ferdie)
+    .await;
+
+    // Run Alice (a core payments domain authority node)
+    let mut bob = domain_test_service::CoreDomainNodeBuilder::new(
+        tokio_handle.clone(),
+        Bob,
+        BasePath::new(directory.path().join("bob")),
+    )
+    .run_relayer()
+    .build_core_payments_node(Role::Authority, &mut ferdie, &alice)
+    .await;
+
+    // Run the cross domain gossip message worker
+    ferdie.start_cross_domain_gossip_message_worker();
+
+    produce_blocks!(ferdie, alice, bob, 3).await.unwrap();
+
+    // Open channel on both sides
+    bob.construct_and_send_extrinsic(pallet_sudo::Call::sudo {
+        call: Box::new(core_payments_domain_test_runtime::RuntimeCall::Messenger(
+            pallet_messenger::Call::initiate_channel {
+                dst_domain_id: DomainId::SYSTEM,
+                params: InitiateChannelParams {
+                    max_outgoing_messages: 100,
+                    fee_model: Default::default(),
```

Comment:
I usually put some number to the fees instead of zero so we may also find other issues regarding the balances